packages/sdk/src/protocols/peer_manager.ts

```diff
@@ -1,113 +1,123 @@
-import { Peer, PeerId } from "@libp2p/interface";
-import { ConnectionManager, getHealthManager } from "@waku/core";
-import { BaseProtocol } from "@waku/core/lib/base_protocol";
-import { IHealthManager } from "@waku/interfaces";
+import { Connection, Peer, PeerId } from "@libp2p/interface";
+import { Libp2p } from "@waku/interfaces";
 import { Logger } from "@waku/utils";
-import { Mutex } from "async-mutex";
```

Comment:
were you able to validate the usage of manual locking instead of using the mutexes lib works well?